lib/tenkft_client.rb

```diff
@@ -0,0 +1,70 @@
+require 'httparty'
+
+class TenkftClient
+  API_BASE_URL = "https://app.10000ft.com/api/v1/"
+  include HTTParty
+
+  attr_reader :cookie
+
+  def initialize
+    raise ArgumentError, 'ENV is missing a key: `10KFT_USERNAME`' unless ENV['10KFT_USERNAME'].present?
+    raise ArgumentError, 'ENV is missing a key: `10KFT_PASSWORD`' unless ENV['10KFT_PASSWORD'].present?
+
+    post_response = HTTParty.post(
+      'https://app.10000ft.com/api/sessions/signin',
+      body: {
+        user_id: ENV['10KFT_USERNAME'],
+        password: ENV['10KFT_PASSWORD']
+      },
+    )
+
+    @cookie = parse_cookie(post_response)[:'tenkft-x']
+  end
+
+  def get_projects(options)
+    get('projects', options)
+  end
+
+  def get_users(options)
+    get('users', options)
+  end
+
+  def get_assignments(user_id, options)
+    get("users/#{user_id}/assignments", options)
+  end
+
+  def get_leave_types(options)
+    get('leave_types', options)
+  end
+
+  def get(endpoint, options)
+    HTTParty.get(
+      API_BASE_URL + endpoint,
+      headers: request_headers,
+      query: options,
+      )
```

Comment:
Super nit-picky: this line has the closing parenthesis indented, but the same style of call in the `post` method below has the closing parenthesis not intended (line 53).